### PR TITLE
Remove ifdef CINT protection in SQL includes

### PR DESCRIPTION
**Pull request by linev**
Long time ago this protection was introduced to avoid parsing of complex SQL includes.
With a cling this protection no longer work properly.
Remove it completely

### CODE REVIEW

File: sql/pgsql/inc/TPgSQLStatement.h [hunk 1]
```diff
@@ -26,6 +28,12 @@ struct PgSQL_Stmt_t {
    PGresult *fRes;
 };
 
+#else
+
+struct PgSQL_Stmt_t {};
```
Comment (bellenot) [inline sql/pgsql/inc/TPgSQLStatement.h]: Is it intentional to remove the `PGresult *fRes;`struct  member?

Comment (linev) [reply sql/pgsql/inc/TPgSQLStatement.h]: It is just for dictionary generation. 
While member is not persistent - we can put anything here to make cling happy.

Comment (bellenot) [reply sql/pgsql/inc/TPgSQLStatement.h]: OK, I see. Thanks

**Comment by Axel-Naumann** [comment]
To understand / reason about what cling / dictionaries see vs. compiler, we should reduce the asymmetry between cling and compiler, not increase it. E.g. hiding the definition of the vector element only for cling can mean code works compiled but not interpreted.

I would be happier of you could replace the `#include`s both for cling and the compiler by forward declarations where possible.

**Comment by linev** [comment]
> I would be happier of you could replace the #includes both for cling and the compiler by forward declarations where possible.

Done.

> we should reduce the asymmetry between cling and compiler, not increase it

Problem to keep external includes in such plugins - one never knows if such includes still there or at same place as they were before. Therefore if possible - I prefer to hide them from interpreter.
Also long time ago cint simply was not able to parse all them.

**Comment by linev** [comment]
@hageboeck 

> LGTM, I would advise squashing on merge.

Why? These are three independent plugins and three commits. 
It would be easier to patch if I will made failure in one

**Comment by linev** [comment]
@bellenot There is Jenkins?

**Comment by bellenot** [comment]
> @bellenot There is Jenkins?

Weird, it disappeared...

**Comment by hageboeck** [comment]
> @hageboeck
> 
> > LGTM, I would advise squashing on merge.
> 
> Why? These are three independent plugins and three commits.
> It would be easier to patch if I will made failure in one

I see. Makes sense now. When you do a shallow review, it looks all the same.

**Comment by Axel-Naumann** [comment]
> Problem to keep external includes in such plugins - one never knows if such includes still there or at same place as they were before. Therefore if possible - I prefer to hide them from interpreter.

If we cannot parse a header at runtime then ROOT is in a broken state (think ACLiC). Hiding this from the interpreter isn't a fix.

**Comment by linev** [comment]
> If we cannot parse a header at runtime then ROOT is in a broken state (think ACLiC). Hiding this from the interpreter isn't a fix.

Then solution - fully remove such ifdefs where possible. 
But then it will be absolutely different PR.
I just tried to repair code and make it working as it was designed.

**Comment by Axel-Naumann** [comment]
> I just tried to repair code and make it working as it was designed.

The design of moving from CINT to cling was to expose to cling what was hidden for CINT.

**Comment by Axel-Naumann** [comment]
> > @bellenot There is Jenkins?

> Weird, it disappeared...

I had problems submitting comments a couple of hours ago - might be related. https://www.githubstatus.com/ doesn't mention anything related :-(

**Comment by linev** [comment]
> The design of moving from CINT to cling was to expose to cling what was hidden for CINT.

Means I remove all ifdefs and let `cling` parse sql headers

**Comment by linev** [comment]
Now all ifdef are removed - hope all platforms will survive

**Comment by linev** [comment]
@Axel-Naumann Is it ok now?

File: sql/sqlite/inc/TSQLiteRow.h [hunk 2]
```diff
@@ -14,12 +14,12 @@
 
 #include "TSQLRow.h"
 
-struct sqlite3_stmt;
+#include <sqlite3.h>
```
Comment (Axel-Naumann) [inline sql/sqlite/inc/TSQLiteRow.h:17]: Isn't a fwd decl enough?

Comment (linev) [reply sql/sqlite/inc/TSQLiteRow.h:17]: No, it has more fancy declaration. Not works this way

**Pull request merged**